filter_grep: grep with arithmetic operators

marcosdiez opened the pull request:
Documentation for https://github.com/fluent/fluent-bit/pull/5449

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Expanded grep filter configuration with numeric comparison parameters and a new "Record Type" column; removed the previous "Default" column.
  * Clarified exclusion behavior when keys are missing or non-numeric and noted that numeric keys don't match regex/exclude.
  * Added an "Arithmetic comparisons" section with examples for equality and combined numeric comparisons (logical AND) and updated configuration examples.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/810?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

marcosdiez committed:
filter_grep: grep with arithmetic operators

Signed-off-by: Marcos Diez <marcos@unitron.com.br>

esmerel committed:
Merge branch 'master' into grep_arithmetic_operators

Signed-off-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>

esmerel commented on pipeline/filters/grep.md:
```diff
@@ -10,11 +10,21 @@ The _Grep_ filter plugin lets you match or exclude specific records based on reg
+| `number_greater_than_or_equal` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is greater than or equal to `NUMBER`. |
+
+
+If you use the number compare parameters with a KEY that does not have a NUMBER as a value, it will be excluded.
```
```suggestion
If you use the number compare parameters with a `KEY` that doesn't have a `NUMBER` as a value, it will be excluded.
```

esmerel reviewed (approved):
This adds a bunch of options, I've updated the table per our style and fixed the merge conflict

esmerel committed:
Update pipeline/filters/grep.md

Signed-off-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>

patrick-stephens reviewed (changes requested):
I think there are some fundamental implementation concerns to resolve but we seem to have removed the `logical_op` key in the docs as well.

eschabell committed:
Merge branch 'master' into grep_arithmetic_operators

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/filters/grep.md:
```diff
@@ -10,11 +10,21 @@ The _Grep_ filter plugin lets you match or exclude specific records based on reg
+| `Regex` | `KEY REGEX` | String | Keep records in which the content of `KEY` matches the regular expression. |
+| `Exclude` | `KEY REGEX` | String | Exclude records in which the content of `KEY` matches the regular expression. |
+| `number_equal` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is equal to `NUMBER`. |
+| `number_not_equal` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is not equal to `NUMBER`. |
```
@patrick-stephens I'm fixing this in the branch and pushing an update here to resolve this.

eschabell replied:
Fixed.

eschabell commented on pipeline/filters/grep.md:
```diff
@@ -14,12 +14,23 @@ The _Grep_ filter plugin lets you match or exclude specific records based on reg
+| `number_less_than_or_equal` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is less than or equal to `NUMBER`. |
+| `number_greater_than` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is greater than the `NUMBER`. |
+| `number_greater_than_or_equal` | `KEY NUMBER` | number | Keep records in which the content of `KEY` is greater than or equal to `NUMBER`. |
 | `regex`      | `KEY REGEX` | Keep records where the content of `KEY` matches the regular expression. | _none_ |
```
fixing.

eschabell commented on pipeline/filters/grep.md:
```diff
@@ -14,12 +14,23 @@ The _Grep_ filter plugin lets you match or exclude specific records based on reg
 | `regex`      | `KEY REGEX` | Keep records where the content of `KEY` matches the regular expression. | _none_ |
 
+
+If you use the number compare parameters with a `KEY` that doesn't have a `NUMBER` as a value, it will be excluded.
+
+If you use `REGEX` or `EXCLUDE` with a number, it will never match it.
+
```
Fixing.

eschabell committed:
docs: pipeline: filters: grep: fix table alignment and add arithmetic examples

  - Restore 5-column table (Key, Value Format, Record Type, Description, Default)
    so exclude, logical_op, and regex rows align correctly with the new number_* rows
  - Fix logical_op row which was rendered into the wrong columns after d7d95864
  - Add 'Arithmetic comparisons' section with single-operator and range examples
    using logical_op: and to address reviewer request for usage examples

  Applies to #810

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell reviewed (approved):
@marcosdiez only waiting on code PR merging.
